synapse/storage/controllers/state.py

```diff
@@ -82,13 +82,15 @@ async def get_state_group_delta(
         return state_group_delta.prev_group, state_group_delta.delta_ids
 
     async def get_state_groups_ids(
```

Comment:
as a random point for future reference: there is a [convention](https://cbea.ms/git-commit/#limit-50) that the subject line of a git commit message should be limited to 50 characters. That helps avoid this sort of thing:

![image](https://user-images.githubusercontent.com/1389908/180444203-6b2e4c6b-096c-4328-8db4-a3d1a4db8a54.png)

There's an art to writing such a pithy summary, of course. In this case I would probably just say:

```
`get_state_groups_ids`: add `await_full_state` parameter
```

... and put "so that it can return partial state" in the body of the commit message.

Reply:
interesting, I wasn't aware that the subject line had a shorter recommended length.

Up until now I've been using a limit of 72 characters in merge commits (and ignoring it in commits that get squashed, like the one above). 50 characters certainly makes things tricky!

Reply:
> ignoring it in commits that get squashed, like the one above

Yeah, the problem comes when we're doing a commit-by-commit review: it's much nicer if there's a pithy summary for each commit. It's certainly not worth stressing overmuch over though.

> 50 characters certainly makes things tricky!

Indeed. github actually seems to truncate at 66 chars, so that seems a more reasonable compromise.